client/src/components/job-applications-view-component/search-component/index.jsx

```diff
@@ -0,0 +1,40 @@
+import { Autocomplete, TextField } from "@mui/material";
+import PropTypes from "prop-types";
+
+function SearchInput({
+  options,
+  label,
+  onChange,
+  selectedValue,
+  getOptionLabel,
+}) {
+  return (
+    <Autocomplete
+      value={selectedValue}
+      onChange={onChange}
+      disablePortal
+      options={options}
+      getOptionLabel={getOptionLabel}
+      sx={{ width: 300 }}
+      renderInput={(props) => (
+        <TextField {...props} size="small" label={label} />
```

Comment:
```suggestion
         /* eslint-disable react/jsx-props-no-spreading */
        <TextField {...props} size="small" label={label} />
```
I encountered this ESLint warning locally ([react/jsx-props-no-spreading](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md)), but since it's related to readability, not functionality, and [MUI's docs](https://mui.com/material-ui/react-autocomplete/) use prop spreading in renderInput, we can safely disable this rule here.